### PR TITLE
feat: portablegabi

### PR DESCRIPTION
**Pull request by weichweich**
## Checklist:

- [ ] I have verified that the code works
- [ ] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/boy-scout-rule/)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

**Comment by flechtnergalani** [comment]
Ponder, my brother: If I allocate a chunk of memory on the heap, but nothing ever reads from it; would I still be justified in calling it "memory" ?

File: runtime/src/portablegabi.rs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+		fn deposit_event<T>() = default;
+
+		/// Updates the attestation
+		pub fn update_accumulator(origin, accumulator: Vec<u8>) -> Result {
+			let attester = ensure_signed(origin)?;
+
+			let counter = if !<AccumulatorCount<T>>::exists(&attester) {
+				0
+			} else {
+				<AccumulatorCount<T>>::get(&attester)
```
Comment (tjwelde) [inline runtime/src/portablegabi.rs:37]: Might be enough to just use this, when it returns 0 for uninitialized ones

File: runtime/src/portablegabi.rs [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+			};
+
+			let next = counter.checked_add(1).ok_or("Overflow increasing accumulator index")?;
+			ensure!(!<AccumulatorList<T>>::exists((attester.clone(), next)),
+					"Inconsistent accumulator counter");
```
Comment (tjwelde) [inline runtime/src/portablegabi.rs:42]: Please use error module

File: runtime/src/portablegabi.rs (hunk 1 above)
Comment (weichweich) [reply runtime/src/portablegabi.rs]: fixed in cb2ed89659a451adee7333d9459bcbb2fe6a8ee1

File: runtime/src/portablegabi.rs (hunk 2 above)
Comment (weichweich) [reply runtime/src/portablegabi.rs]: fixed in cb2ed89659a451adee7333d9459bcbb2fe6a8ee1

**Pull request merged**